cli/src/cli.rs

```diff
@@ -58,6 +64,120 @@ pub enum Subcommand {
 
     /// Revert the chain to a previous state.
     Revert(sc_cli::RevertCmd),
+
+    #[clap(subcommand)]
+    FixBabeEpoch(FixEpochSubCommand),
+}
+
+#[derive(Debug, clap::Subcommand)]
+pub enum FixEpochSubCommand {
+    Dump(FixEpochDumpCommand),
+    Override(FixEpochOverrideommand),
+}
+
+impl FixEpochSubCommand {
+    /// Run the revert command
+    pub fn run<C: SubstrateCli>(&self, cli: &C) -> Result<()> {
+        match self {
+            FixEpochSubCommand::Dump(cmd) => cmd.run(cli),
+            FixEpochSubCommand::Override(cmd) => cmd.run(cli),
+        }
+    }
+}
+
+#[derive(Debug, clap::Parser)]
+pub struct FixEpochDumpCommand {
+    #[allow(missing_docs)]
+    #[clap(flatten)]
+    pub shared_params: SharedParams,
+
+    #[allow(missing_docs)]
+    #[clap(flatten)]
+    pub pruning_params: PruningParams,
+}
+const BABE_EPOCH_CHANGES_KEY: &[u8] = b"babe_epoch_changes";
+use codec::{Decode, Encode};
+use sc_consensus_babe::Epoch;
+use sc_consensus_epochs::EpochChangesFor;
+
+impl FixEpochDumpCommand {
+    pub fn run<C: SubstrateCli>(&self, cli: &C) -> Result<()> {
+        let runner = cli.create_runner(self)?;
+        runner.sync_run(|mut config| {
+            let components = new_partial::<
+                chainx_runtime::RuntimeApi,
+                chainx_executor::ChainXExecutor,
+            >(&mut config)?;
+            let client = components.client;
+            let bytes = client
+                .get_aux(BABE_EPOCH_CHANGES_KEY)
+                .expect("Access DB should success")
+                .expect("value must exist");
+            let hex_str = hex::encode(&bytes);
+            println!("{}", hex_str);
+            let epoch: EpochChangesFor<chainx_runtime::Block, Epoch> =
+                codec::Decode::decode(&mut &bytes[..]).expect("Decode must success");
+            println!("epoch: {:?}", epoch);
+            Ok(())
+        })
+    }
+}
+
+#[derive(Debug, clap::Parser)]
+pub struct FixEpochOverrideommand {
+    #[allow(missing_docs)]
+    #[clap(flatten)]
+    pub shared_params: SharedParams,
+
+    #[allow(missing_docs)]
+    #[clap(flatten)]
+    pub pruning_params: PruningParams,
+
+    #[clap(long)]
+    pub bytes: String,
```

Comment:
advice to change the `String` to `Bytes` at here, and provide `str2hex` function, rather then convert the input to bytes in `run` function.

Reply:
The dumped data is a string in hex format, and the data format specified by `override` is also hex format.